bundles/org.openhab.binding.cbus/src/main/java/org/openhab/binding/cbus/handler/CBusNetworkHandler.java

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.cbus.handler;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.cbus.internal.CBusNetworkConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.daveoxley.cbus.CGateException;
+import com.daveoxley.cbus.CGateSession;
+import com.daveoxley.cbus.Network;
+import com.daveoxley.cbus.Project;
+
+/**
+ * The {@link CBusNetworkHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Scott Linton - Initial contribution
+ */
+@NonNullByDefault
+public class CBusNetworkHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(CBusNetworkHandler.class);
+    private @Nullable CBusCGateHandler bridgeHandler = null;
+    private @Nullable CBusNetworkConfiguration configuration = null;
+    private @Nullable Network network = null;
+    private @Nullable Project projectObject = null;
+    private @Nullable ScheduledFuture<?> initNetwork = null;
+    private @Nullable ScheduledFuture<?> networkSync = null;
+
+    public CBusNetworkHandler(Bridge thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initialize --");
+        configuration = getConfigAs(CBusNetworkConfiguration.class);
```

Comment:
This is to deal with the null warnings
```suggestion
        CBusNetworkConfiguration configuration = getConfigAs(CBusNetworkConfiguration.class);
        this.configuration = configuration;
```

Reply:
I dont see any null warnings and since configuration isn't used and is just assigned to here i dont see why this is necessary?

Reply:
If it isn't necessary then remove it

Reply:
Ah, sorry got ahead of myself, I saw that you were logging it and just applied my general rule of "If a nullable field is used locally then it needs a local variable". But you are correct, a local variable isn't necessary in this instance.